### PR TITLE
[USER32]  Fix wrong return from LB_GETLISTBOXINFO

### PR DESCRIPTION
**Pull request by zecarlos1957**
## Purpose

 win32ss/user/user32/controls/listbox.c   
      Fixed wrong return from LB_GETLISTBOXINFO in case LBS_MULTICOLUMN
  According to Microsoft the message LB_GETLISTBOXINFO should return the number of visible items in the column. In case of a list style LBS_MULTICOLUMN the returned value(line 3045), has no meaning since the calculation multiplies the number of items in a column by its width in pixels.

win32ss/user/ntuser/window.c Line _3422   must match listbox.c L_3045 
 (see window.c line 3420)

### CODE REVIEW

**Comment by HBelusca** [comment]
To be checked also wrt. corresponding wine code, cc @jimtabor .

**Comment by HBelusca** [comment]
What happened @zecarlos1957 ?

**Comment by zecarlos1957** [comment]
Maybe I misunderstood your comment. If you find this PR useful, I'll reopen it.

**Comment by KyleKatarn31** [comment]
Has the change been pushed to WineHQ ?

Screenshot pre/post PR ?

File: win32ss/user/ntuser/window.c [hunk 1]
```diff
@@ -3419,7 +3419,11 @@ NtUserGetListBoxInfo(
       LB_DESCR *descr = ((PWND2LB)Wnd)->pLBiv;
       // See Controls ListBox.c:LB_GETLISTBOXINFO must match...
       if (descr->style & LBS_MULTICOLUMN) //// ReactOS
-         Ret = descr->page_size * descr->column_width;
+#ifdef __REACTOS__
+         Ret = descr->page_size * (descr->width / descr->column_width);
+#else
+         Ret = descr->page_size * descr->width;
```
Comment (katahiromz) [inline win32ss/user/ntuser/window.c:3425]: Keep original line

File: win32ss/user/user32/controls/listbox.c [hunk 2]
```diff
@@ -3042,7 +3042,11 @@ LRESULT WINAPI ListBoxWndProc_common( HWND hwnd, UINT msg,
 
     case LB_GETLISTBOXINFO:
         if (descr->style & LBS_MULTICOLUMN) //// ReactOS
-           return descr->page_size * descr->column_width;
+#ifdef __REACTOS__
+           return descr->page_size * (descr->width / descr->column_width);
+#else
+           return descr->page_size * descr->width;
```
Comment (katahiromz) [inline win32ss/user/user32/controls/listbox.c:3048]: Keep original line

File: win32ss/user/ntuser/window.c (hunk 1 above)
Comment (zecarlos1957) [reply win32ss/user/ntuser/window.c]: @katahiromz  my mistake. Thanks

**Comment by zecarlos1957** [comment]
Window says: LB_GETLISTBOXINFO should return the number of items per column in a specified list box.

File: win32ss/user/user32/controls/listbox.c [hunk 3]
```diff
@@ -3041,9 +3041,11 @@ LRESULT WINAPI ListBoxWndProc_common( HWND hwnd, UINT msg,
         return LB_OKAY;
 
     case LB_GETLISTBOXINFO:
+#ifndef __REACTOS__
         if (descr->style & LBS_MULTICOLUMN) //// ReactOS
```
Comment (binarymaster) [inline win32ss/user/user32/controls/listbox.c:3045]: What I see here is code inside `#ifndef __REACTOS__` block have `//// ReactOS` comment, this looks quite strange, where this code block comes from? I can't find it in Wine repository.

Comment (zecarlos1957) [reply win32ss/user/user32/controls/listbox.c:3045]: @binarymaster 
    case LB_GETLISTBOXINFO:
        if (descr->style & LBS_MULTICOLUMN) //// ReactOS
           return descr->page_size * descr->column_width;
        else
           return descr->page_size;
 
This is the original code in win32ss/user/user32/controls/listbox.c,,  but in fact as I tested  on Windows, LB_GETLISTBOXINFO always returns the number of items in a column, whether MULTICOLUMN type or not.

Comment (binarymaster) [reply win32ss/user/user32/controls/listbox.c:3045]: So... why `#ifndef __REACTOS__` ?

Comment (zecarlos1957) [reply win32ss/user/user32/controls/listbox.c:3045]: <return descr-> page_size * descr-> column_width> (Wrong value)
only returns when __REACTOS__ is not defined. Correct?

Comment (zecarlos1957) [reply win32ss/user/user32/controls/listbox.c]: @binarymaster  Now is equal to WINE code

**Pull request merged**